src/robot/variables/replacer.py

```diff
@@ -139,31 +139,31 @@ def _get_variable_item(self, match, value):
         for item in match.items:
             if is_dict_like(value):
                 value = self._get_dict_variable_item(name, value, item)
-            elif is_list_like(value):
-                value = self._get_list_variable_item(name, value, item)
+            elif is_iterable(value):
+                value = self._get_iterable_variable_item(name, value, item)
             else:
                 raise VariableError(
-                    "Variable '%s' is %s, not list or dictionary, and thus "
+                    "Variable '%s' is %s, not a dictionary or iterable, and thus "
                     "accessing item '%s' from it is not possible."
                     % (name, type_name(value), item)
                 )
             name = '%s[%s]' % (name, item)
         return value
 
-    def _get_list_variable_item(self, name, variable, index):
+    def _get_iterable_variable_item(self, name, variable, index):
         index = self.replace_string(index)
         try:
-            index = self._parse_list_variable_index(index, name[0] == '$')
+            index = self._parse_iterable_variable_index(index, name[0] == '$')
         except ValueError:
-            raise VariableError("List '%s' used with invalid index '%s'."
+            raise VariableError("Iterable '%s' used with invalid index '%s'."
```

Comment:
This is another place where we could enhance the error message. The biggest problem is that you end up here if you have data like `${xpath}[predicate]` and didn't mean to use the item access syntax. Something like "To use '[%s]' as a literal value, it needs to be escaped like '\\[%s]'." might be a good addition. This was discussed already above, and possibly both situations could be handled by same code.

Another issue that the word "iterable" may not be that familiar for all users. Not sure what else to use, though, as "list" or even "list-like" aren't accurate here anymore.

Reply:
I'm changing "iterable" to "sequence" and extending the error message as you suggested.